UTAT_Compression_Algorithm/C/helperlib.c

```diff
@@ -0,0 +1,51 @@
+#include <math.h>
+#include <gsl/gsl_vector.h>
+
+/*
+This file implements helper functions used in the compression and 
+decompression steps.
+*/
+
+//Returns the sign of an integer, with 0 returned as a positive
+int sign(int x) {
+    if (x >= 0) return 1;
+    else return -1;
+}
+
+double clamp(double x, double min, double max){
+    double result = x;
+    if (x < min)
+        result = min;
+    else if (x > max)
+        result = max;
+    return result;
+}
+//Returns a binary array from a decimal
+//Binary arrays are stored as GSL integer vectors
+gsl_vector_int* dec_to_bin(int num, int width){
+    gsl_vector_int* bin = gsl_vector_int_alloc(width);
+
+    int i = 0;
+    while (i < width){
+        
+        gsl_vector_int_set(bin, i, num %2);
+        
+        num = num / 2;
+        
+        i++;
```

Comment:
something feels off here... Like num is defined as an int but you're using num/2. But we had double values before. Just double check this. Also isn't the int already a 32 or 64-bit block that stores the array you want? Because we're in C now, instead of Python, I have a feeling this stuff can be completely removed.